### DISCUSSION TITLE
Micropython on STM32F405RGT6 Increase File System Allotment

### DISCUSSION THREAD

**Discussion by AngeloFatica** [micropython/micropython#11407] [STM32 / Pyboard]
Forgive me if there is some discussion on this already, there is a lot to take in here and I could not find an answer myself. Is there a way to "trick" the memory on lets say the pyboard for instance to allow for less flash storage "text" and more file system storage? Ive tried playing around with different builds and adjusting the linker scripts to no successful build. Has anyone been able to do this successfully? I can just freeze my code anyways but I am curious about this. Thanks in advance.

**Comment by jimmo** [comment]
@AngeloFatica 
There's no "trick", but the issue is that the flash is arranged in sectors with different sizes, but the flash cache is limited to the size of CCRAM, so we can only use the first 64kiB in each sector for the filesystem. The default configuration is to use sectors 1,2,3,4 (which are 16kiB,16kiB,16kiB,64kiB respectively) but all other sectors in the F405 are larger (128kiB).

So each additional 64kiB you add to the filesystem takes a 128kiB sector from the firmware.

I don't know what you've tried (perhaps you could share your changes here?) but the main thing to remember is that the changes you make to the linker script must match the sector sizes.
